src/ebonite/repository/metadata/local.py

```diff
@@ -366,6 +379,8 @@ def update_model(self, model: Model) -> Model:
         return model
 
     def delete_model(self, model: Model):
+        if model.images:
```

Comment:
I recommend using `self.get_images` here as `Model.images` is not guaranteed to reflect DB state.